Hunting Queries/W3CIISLog/PotentialWebshell.txt

```diff
@@ -0,0 +1,19 @@
+
+//Name: Web shell Detection 
+//Description: Web shells are script that when uploaded to a web server can be used for remote administration . Attackers often use web shells to obtain unauthorized access, escalate //privilege as well as further compromise the environment. The query detects web shells that use GET requests by keyword searches in URL strings. This is based out of sigma rules described //here (https://github.com/Neo23x0/sigma/blob/master/rules/web/web_webshell_keyword.yml). There could be some web sites like wikis with articles on os commands and pages that include the os //commands in the URLs that might cause FP.
+//
+// Id: cc087e7c-4db0-4bf9-9e48-287a9c9c3fbc
+//
+//
+// DataSource: W3CIISLog
+//
+// Tactics: #LateralMovement, #PrivilegeEscalation ,#C2
+//
+let timeFrame = ago(1d);
+W3CIISLog
+| where TimeGenerated >= timeFrame
+| where csMethod == "GET" 
+| where ( csUriQuery contains "whoami" or csUriQuery contains "net%20user" or csUriQuery contains "cmd%20/c%20" ) or 
```

Comment:
For the "whoami" lookup, switch contains to a has, it is more efficient.  The others will require contains because they have potential indexing characters in them.